# Тести для малих вибірок та пропорцій



**Завдання 1**. E-commerce компанія після редизайну сайту підозрює, що **середній час до покупки (time-to-purchase)** користувачів **збільшився**.
Історично середній час від першого заходу на сайт до покупки становив $\mu_0 = 123.7$ хвилин.

Команда аналітиків випадково обрала дані **7 покупців після редизайну**:
`128, 135, 121, 142, 126, 151, 123`

З допомогою наявних даних зʼясуйте, чи збільшився середній час покупки після редизайну.

Для цього виконайте наступні 6 кроків. Правильне виконання кожного з кроків оцінюється в 1 бал.

1. Запишіть нульову та альтернативну гіпотези і визначте тип тесту.
2. Обчисліть вибіркові статистики: $\bar x$, $s$, $n$.
3. Оберіть тип тесту та виконайте його (знайдіть критичне значення тесту, статистику тесту та р-значення) будь-яким пасуючим способом, щоб перевірити гіпотезу на рівні значущості $\alpha = 0.10$.
4. Прийміть рішення, чи відхиляєте ви гіпотезу $H_0$ використовуючи p-value.
5. Напишіть висновок: чи справді редизайн сайту подовжив час до покупки?
6. Чи зміниться ваше рішення при зміні рівня значущості на $\alpha = 0.05$.


In [3]:
import pandas as pd
import numpy as np
from scipy import stats
from statsmodels.stats import weightstats as stests
import statsmodels.api as sm
from statsmodels.stats.proportion import proportions_ztest, proportion_confint
import pandas.util.testing as tm

In [8]:
sample = np.array([128, 135, 121, 142, 126, 151, 123])
mu0 = 123.7
alpha = 0.10

print("-----1-----")
print("H₀: μ = 123.7")
print("H₁: μ > 123.7")
print("Правосторонній тест")
print()

print("-----2-----")
x_bar = np.mean(sample)
s = np.std(sample, ddof=1)
n = len(sample)
print(f"Вибіркове середнє: {x_bar:.2f}")
print(f"Вибіркове стандартне відхилення: {s:.2f}")
print(f"Розмір вибірки: {n}")

-----1-----
H₀: μ = 123.7
H₁: μ > 123.7
Правосторонній тест

-----2-----
Вибіркове середнє: 132.29
Вибіркове стандартне відхилення: 10.98
Розмір вибірки: 7


**Завдання 2.**

До спеціальної рекламної кампанії **23%** дорослих упізнавали логотип компанії. Після завершення кампанії відділ маркетингу провів опитування: з **1200** випадково відібраних дорослих **311** упізнали логотип.

Перевірте на рівні значущості **3%** ($\alpha=0.03$), чи дають ці дані достатні підстави стверджувати, що **тепер більше ніж 23%** дорослих упізнають логотип компанії. Для розвʼязку використовуйте бібліотеку `statsmodels`.

Зробіть висновок, чи зросла впізнаваність логотипу.

Додатково, обчисліть довірчий інтревал на заданому рівні значущості і проінтерпретуйте текстом - як він додатково пояснює прийняте нами рішення?

In [7]:
count = 311 
nobs = 1200
p0 = 0.23
alpha = 0.03

print("H₀: p = 0.23")
print("H₁: p > 0.23")
print("Правосторонній тест")
print()

stat, pval = proportions_ztest(count, nobs, p0, alternative='larger')
confint = proportion_confint(count, nobs, alpha, method='normal')

print("Тест")
print(f"Z-статистика = {stat:.3f}")
print(f"p-value = {pval:.4f}")
print(f"97% довірчий інтервал: {confint}")

print()
print("ВИСНОВОК")
if pval < alpha:
    print(f"При alpha={alpha} ми відхиляємо H₀, і можемо стверджувати що впізнаваність логотипу статистично значущо зросла")
    print("і тепер становить більше ніж 23%")
else:
    print(f"При alpha={alpha} ми НЕ ВІДХИЛЯЄМО H₀, у нас немає підстав стверджувати, що впізнаваність логотипу зросла")

H₀: p = 0.23
H₁: p > 0.23
Правосторонній тест

Тест
Z-статистика = 2.306
p-value = 0.0106
97% довірчий інтервал: (0.23171700302179205, 0.28661633031154127)

ВИСНОВОК
При alpha=0.03 ми відхиляємо H₀, і можемо стверджувати що впізнаваність логотипу статистично значущо зросла
і тепер становить більше ніж 23%


Ми маємо достатньо статистичних доказів (p<0.03), щоб відкинути нульову гіпотезу та визнати рекламну кампанію ефективною. Впізнаваність логотипу реально зросла. Довірчий інтервал від 23.17% до 28.66% чітко вказує на позитивну динаміку, адже навіть мінімально прогнозоване значення є вищим за стартовий показник у 23%.